metricbeat/tests/system/test_autodiscover_jolokia.py

```diff
@@ -0,0 +1,54 @@
+import os
+import metricbeat
+import unittest
+from nose.plugins.attrib import attr
+
+
+class Test(metricbeat.BaseTest):
+
+    COMPOSE_SERVICES = ['jolokia']
+
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    def test_jolokia_jmx(self):
```

Comment:
This is the only test really covering this feature by now, and is under metricbeat. This is not ideal, but it should be enough to detect big issues in its main use case.
Big part of the code is dependant of the discovery protocol and for that we'd need scenarios like the ones we have in metricbeat. I'd leave for the future further tests, like mocking all the networking-dependant part, or preparing a more general testing scenario under libbeat.